core/base/src/TApplication.cxx

```diff
@@ -628,20 +630,423 @@ TApplication::EExitOnException TApplication::ExitOnException(TApplication::EExit
    return old;
 }
 
-////////////////////////////////////////////////////////////////////////////////
-/// Print help on interpreter.
+/////////////////////////////////////////////////////////////////////////////////
+/// The function generates and executes a command that loads the Doxygen URL in
+/// a browser. It works for Mac, Windows and Linux. In the case of Linux, the
+/// function also checks if the DISPLAY is set. If it isn't, a warning message
+/// and the URL will be displayed on the terminal.
+///
+/// \param[in] url <web page to be displayed in a browser
 
-void TApplication::Help(const char *line)
+void TApplication::OpenInBrowser(const TString &url)
+{
+   // Commands for opening a browser in Linux, Windows and Mac
+   TString cLinux("xdg-open ");
+   TString cWindows("start ");
+   TString cMac("open ");
+   // We check what operating system the user has
+   if (gSystem->InheritsFrom("TMacOSXSystem")) {
+      // We generate the full command and execute it
+      cMac.Append(url);
+      gSystem->Exec(cMac.Data());
+   } else if (gSystem->InheritsFrom("TWinNTSystem")) {
+      cWindows.Append(url);
+      gSystem->Exec(cWindows.Data());
+   } else {
+      // For Linux we check if the DISPLAY is set
+      TString displayhandle(gSystem->Getenv("DISPLAY"));
+      if ((displayhandle.Sizeof()) > 1) {
+         // If the DISPLAY is set it will open the browser
+         cLinux.Append(url);
+         gSystem->Exec(cLinux.Data());
+      } else {
+         // Else the user will have a warning and the URL in the terminal
+         Warning("Help", "The DISPLAY is not set!");
+         std::cout << url.Data() << std::endl;
+      }
+   }
+}
+namespace {
+static TString URLGenerator(TString ClassName)
+{
+   // Generates a URL address for the class (classname).
```

Comment:
Can you move this documentation outside this function and give it doxygen formatting?